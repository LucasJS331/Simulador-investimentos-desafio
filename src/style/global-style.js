import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
    *{
        box-sizing: border-box;
        padding: 0;
        margin: 0;
        font-family: 'arial';
    }
    html{
        font-size: 62.5%;
        scroll-behavior: smooth;
    }
    body{
        font-size: 1.6rem;
    }
    input {
      display: block;
      outline: none;
      border: none;
      border-bottom: 1px solid black;
      background-color: transparent;
      padding: 0.5rem;
      width: 100%;
    }

`;
