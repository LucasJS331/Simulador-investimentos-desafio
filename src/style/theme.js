export const theme = {
  colors: {
    primary_color: '#E5E8EB',
    secondary_color: '#FF7D48',
    secondary_color_hover: '#E16736',
  },
  media: {
    default: '(max-width: 830px)',
  },
};
