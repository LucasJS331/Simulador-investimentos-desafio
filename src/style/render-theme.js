import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { FinantialContext } from '../context/finantial-context';
import { theme } from './theme';

export const renderTheme = (children) => {
  return render(
    <FinantialContext>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </FinantialContext>,
  );
};
