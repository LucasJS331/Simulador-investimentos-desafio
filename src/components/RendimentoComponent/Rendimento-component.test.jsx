import { fireEvent, screen } from '@testing-library/react';
import { Rendimento } from '.';
import { renderTheme } from '../../style/render-theme';

describe('<Rendimento />', () => {
  it('Should render rendimento component', () => {
    const { container } = renderTheme(<Rendimento />);
    expect(container).toMatchSnapshot();
  });
  it('Should change option', () => {
    renderTheme(<Rendimento />);
    const liquido_option = screen.getByText(/liquido/i);
    const bruto_option = screen.getByText(/bruto/i);

    fireEvent.click(liquido_option);
    expect(liquido_option).toHaveClass('rendimento-br-10');
    fireEvent.click(bruto_option);
    expect(liquido_option).toHaveClass('rendimento-bg-br-10');
    expect(bruto_option).toHaveClass('rendimento');
  });
});
