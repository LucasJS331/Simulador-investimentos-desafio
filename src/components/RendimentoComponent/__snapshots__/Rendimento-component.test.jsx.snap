// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rendimento /> Should render rendimento component 1`] = `
<div>
  <div
    class="sc-bdvvtL IpnXL"
  >
    <h3
      id="rendimento-title"
    >
      Rendimento
    </h3>
    <a
      class="rendimento"
      href="#"
    >
      Bruto
    </a>
    <a
      class="rendimento-bg-br-10"
    >
      Liquido
    </a>
  </div>
</div>
`;
